Improve ReadMes and add generate scenario script

IvanPombo opened the pull request:
This PR improves the ReadMe's with more specific example of low-level calls in order to get a nice visual, and with the construction of the scenario itself.

Then, I have also added a dataset generation script. Already tested by launching 100 simulations, where 51 failed due to input flaws.

IvanPombo committed:
Improve README and add dataset generation

Add main README with low-level examples

Improve ReadMes and Add dataset generation

sarmento commented on README.md:
````diff
@@ -5,9 +5,69 @@
+```python
+import inductiva
+
+input_dir = inductiva.utils.files.download_from_url(
````
can this be part of the repo now?
Can we avoid having to download things?

IvanPombo replied:
We still want to be able to download files from bucket to demo simulators in other documentation sites.

sarmento replied:
But this demos should be as simple as possible.
If we can avoid importing inductiva utils it would be better.
If the files are in the repo, then the user will just need to point to them (or we can just have a relative path somewhere).

IvanPombo replied:
Will do that. However, keeping the download for now. We will see how the documentation evolves before removing.

sarmento commented on README.md:
````diff
@@ -5,9 +5,69 @@
+# Wait for the simulation to finish
+task.wait()
+task.download_outputs()
+```
````
show me what I would see (or how would I see it or where I would see it).

sarmento commented on README.md:
```diff
@@ -5,9 +5,69 @@
+import inductiva
+
+# Download the configuration files into a folder
+input_dir = inductiva.utils.files.download_from_url(
```
can some of this be part of the repo?
We could potentially have a metaball if needed for IP reasons.

IvanPombo replied:
You can avoid it for sure, we just need to hold the data elsewhere. All of the input examples we are using are Public.

sarmento commented on README.md:
```diff
@@ -5,9 +5,69 @@
 Here, we explore Smoothed Particle Hydrodynamics (SPH) with two simulators available via **Inductiva API** - [SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and [DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics). 
 
 In this repository, we make a tour of **Inductiva API**:
+- [Run a simulation](README.md#running-a-simulation);
```
Running a simulation by directly calling the Inductiva API (?)

sarmento commented on README.md:
```diff
@@ -5,9 +5,69 @@
 
 In this repository, we make a tour of **Inductiva API**:
+- [Run a simulation](README.md#running-a-simulation);
 - [Create a scenario](scenario/README.md);
```
a custom Python class to run a specific simulation scenario (?)

IvanPombo replied:
Going to avoid the Python, that is the tool not the core.

sarmento commented on README.md:
```diff
@@ -5,9 +5,69 @@
 In this repository, we make a tour of **Inductiva API**:
+- [Run a simulation](README.md#running-a-simulation);
 - [Create a scenario](scenario/README.md);
 - [Generate a dataset via scenario](dataset/README.md).
```
Use that Python class to generate a synthetic dataset at scale (?)

sarmento reviewed (commented):
Just a few comments to help converge on the terminology/story telling...

IvanPombo committed:
Address Sarmento comments to improve storytelling

IvanPombo committed:
Clean dataset generation script

sarmento commented on README.md:
````diff
@@ -5,8 +5,73 @@
+task.download_outputs()
+```
 
 ### Install
````
One more note: I believe that the first page should be more compact. Maybe we should move these two examples under a different document and have a link such as "The Basics: How to call SplishSplash or DualSPhisics via de API". This should be more of a reminder, since part of this explanation will already be in the core package. The same thing for the instalation. It should not occupy a significant chunk of the 1st page. 
Ideally, the first page should be much more visual. Look at this as a reference:

https://github.com/InteractiveComputerGraphics/SPlisHSPlasH

It opens with an animation. The animation there makes the whole repo self-explanatory just by itself.

sarmento reviewed (commented):
I few more notes on the storytelling.
(this is a great piece of work, btw. It will have huge impact).

IvanPombo committed:
Clean the ReadMes and add toy simulation

IvanPombo committed:
Remove scenario folder and clean lib based on mixins

Add sph-demos

IvanPombo committed:
Clear pylint errors

IvanPombo committed:
Final yapf correction.

IvanPombo committed:
Push Fluid Block read me changes

sergiomsantos commented on sph/models.py:
```diff
@@ -0,0 +1,64 @@
+    def __post_init__(self):
+        """Validates the input parameters after initialization."""
+
+        assert 400 <= self.density <= 2000, '`density` should be in [400, 2000]'
```
```suggestion
        assert 400 <= self.density <= 2000, '`density` should be in [400, 2000] range'
```

sergiomsantos commented on sph/scenarios.py:
```diff
@@ -0,0 +1,112 @@
+    time_step: float = 0.001
+
+    def to_dict(self):
+        """Returns a dictionary representation of the model."""
```
```suggestion
        """Returns a dictionary representation of the simulation parameters."""
```

sergiomsantos commented on sph/scenarios.py:
```diff
@@ -0,0 +1,112 @@
+        """
+        self.fluid_block = fluid_block
+
+    def get_commands(self):
+        commands = [{
+            "cmd": "gencase fluid_block fluid_block -save:all",
+            "prompts": []
+        }, {
+            "cmd":
+                "dualsphysics fluid_block fluid_block -dirdataout data -svres",
+            "prompts": []
+        }, {
+            "cmd": "partvtk -dirin fluid_block/data -savevtk vtk/PartFluid"
+                   " -onlytype:-all,+fluid",
+            "prompts": []
+        }]
```
```suggestion
    def get_commands(self):
        commands = [{
            "cmd": "gencase fluid_block fluid_block -save:all",
            "prompts": []
        }, {
            "cmd": "dualsphysics fluid_block fluid_block -dirdataout data -svres",
            "prompts": []
        }, {
            "cmd": "partvtk -dirin fluid_block/data -savevtk vtk/PartFluid -onlytype:-all,+fluid",
            "prompts": []
        }]
```

sergiomsantos commented on sph/post_processing.py:
```diff
@@ -85,3 +86,30 @@ def create_movie_from_vtk(vtk_output_dir: str,
         clip.write_videofile(movie_path)
+
+
+def render(source_dir,
```
swap the functions:
`render` calls `create_movie_from_vtk` so it is natural that render comes first.

IvanPombo replied:
I would assume the reverse, since I need the `create_movie_from_vtk` to create `render`. But I guess for persons checking the code, the perception changes.

sergiomsantos commented on sph/__init__.py:
```diff
@@ -0,0 +1,5 @@
+#pylint: disable=missing-module-docstring
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
```
let's move this section to the demo scripts.
Log config should be made there and not on the src lib.

sergiomsantos commented on generate_dataset.py:
```diff
@@ -0,0 +1,70 @@
+"""Generate a dataset of FluidBlock simulations.
+
+The dataset contains variations of the viscosity, position, dimensions
+and initial velocity of the block of fluid as it is released in the tank.
```
```suggestion
and initial velocity of the block of fluid as it is released into the tank.
```

sergiomsantos reviewed (commented):
still need to go over the *md files.
Looking good.

IvanPombo committed:
Improve ReadMes and Code Flow with latest refactor of scenarios

Note: generate_dataset.py is still being converted to the latest setup.

Change generate_dataset to comply with the changes

Address yapf formatting

Delete cpp-package.yml

Fix yapf formatting

Update markdown and gifs

Update fluid_block.gif to higher resolution

Improve Dataset Readme

sergiomsantos commented on lib/scenarios.py:
```diff
@@ -0,0 +1,112 @@
+        commands = [{
+            "cmd": "gencase fluid_block fluid_block -save:all",
+            "prompts": []
+        }, {
+            "cmd":
+                "dualsphysics fluid_block fluid_block -dirdataout data -svres",
+            "prompts": []
+        }, {
+            "cmd":
+                "partvtk -dirin fluid_block/data -savevtk vtk/PartFluid -onlytype:-all,+fluid",
+            "prompts": []
+        }]
```
```suggestion
        }, {
            "cmd": "dualsphysics fluid_block fluid_block -dirdataout data -svres",
            "prompts": []
        }, {
            "cmd": "partvtk -dirin fluid_block/data -savevtk vtk/PartFluid -onlytype:-all,+fluid",
            "prompts": []
        }]
```

IvanPombo replied:
Yapf doesn't allow for this break, I have tried, but it always complains... We could remove the formatting files from these repos to simplify.

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,21 @@
-In this repository, we make a tour of **Inductiva API**:
-- [Create a scenario](scenario/README.md);
-- [Generate a dataset via scenario](dataset/README.md).
+With [**Inductiva API**](https://github.com/inductiva/inductiva/tree/main) exploring Smoothed Particle Hydrodynamics (SPH) simulations was never easier. Users can launch the simulations they are familiar with through a simple Python interface. 
```
```suggestion
With [**Inductiva API**](https://github.com/inductiva/inductiva/tree/main), exploring Smoothed Particle Hydrodynamics (SPH) simulations was never easier.
Users can launch the simulations they are familiar with through a simple Python interface, without having
to worry about downloading/compiling and installing simulation packages and managing computational resources.
```

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,21 @@
+With [**Inductiva API**](https://github.com/inductiva/inductiva/tree/main) exploring Smoothed Particle Hydrodynamics (SPH) simulations was never easier. Users can launch the simulations they are familiar with through a simple Python interface. 
 
-### Install
+Two open-source SPH simulators are available via **Inductiva API** - [SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and [DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics). On top of them, users can create their custom scenarios and explore a design space with simplicity.
```
```suggestion
Two open-source SPH simulators are available via **Inductiva API** - [SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and [DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics). Users can use use them directly
or leverage the API to create custom scenarios and explore design spaces with simplicity.
```

sergiomsantos commented on README.md:
````diff
@@ -1,31 +1,21 @@
-```bash
-git clone https://github.com/inductiva/inductiva-sph-demos.git
-```
+The **Inductiva API** guide for SPH simulations is the following:
+- [Running a simulation](SPH_SIMULATIONS.md#running-a-simulation);
+- [Create a custom simulation scenario](FLUID_BLOCK.md);
+- [Generate a synthetic dataset at scale](DATASET.md).
````
Installation should be the first bullet point. All installation related stuff should go in that section

```suggestion
In this tour, you will learn how to integrate **Inductiva API**  SPH simulation workflows through several examples:
- [How to use this repository and install any dependencies](INSTALL.md);
- [Run a simulation using straightfoward API calls](SPH_SIMULATIONS.md);
- [Leverage the API to create a custom simulation scenario](FLUID_BLOCK.md);
- [Generate a synthetic dataset at scale](DATASET.md).
```

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,21 @@
+Two open-source SPH simulators are available via **Inductiva API** - [SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and [DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics). On top of them, users can create their custom scenarios and explore a design space with simplicity.
 
-To start experimenting with these scenarios you need to clone the public GitHub repository
+First, don't forget to `pip install inductiva` and then you can follow the [install instructions](INSTALL.md) for this repository to get started.
```
This should go inside INSTALL.md.

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,21 @@
+- [Generate a synthetic dataset at scale](DATASET.md).
 
-and install the `inductiva` package and other extra dependencies that are useful for this scenario.
+In this tour, you will learn how to integrate **Inductiva API**  SPH simulation workflows through several examples.
```
this line can be removed.

sergiomsantos commented on README.md:
````diff
@@ -1,31 +1,21 @@
-```
-
-To further learn about the `inductiva` package, check the [Inductiva API documentation](https://github.com/inductiva/inductiva/wiki).
+To learn more about the `inductiva` package, check the [Inductiva API documentation](https://github.com/inductiva/inductiva/wiki).
````
maybe add a link to wikipedia on SPH?

sergiomsantos commented on INSTALL.md:
````diff
@@ -0,0 +1,15 @@
+### Install
+
+After having installed `inductiva` package with pip, the next step is to clone this repository:
+
+```bash
+git clone https://github.com/inductiva/inductiva-sph-demos.git
+```
+
+Then, for analysis of the results we suggest the installation of a few extra dependencies from within the `inductiva-sph-demos`:
+```bash
+pip install -r requirements.txt
+```
+
+To run visualizations you may be required to install `ffmpeg`, if it is not already installed in your setup. You can download it from
+[here](https://ffmpeg.org/download.html), or if using Anaconda to manage the environment install it directly with `conda install -c conda-forge ffmpeg`.
````
````suggestion
### Install

#### `inductiva` client

The `inductiva` package is required to run the examples in this repository.
It provides the user with mechanisms to run simulations at scale, without
having to be worried about installing simulation packages and managing
physical resources. Users need only to handle the science behind the problem
and configure the simulation using the packages they know and love.
You handle the science, Inductiva manages the rest.

The `inductiva` package is easily installed using `pip`:

```bash
$ pip install inductiva
```

A successful installation can be tested using:

```bash
$ python -c 'import inductiva; print(inductiva.__version__)'
<some version number printed here> 
```

#### demo repo

Having the `inductiva` client available, the next step is to clone this repository:

```bash
$ git clone https://github.com/inductiva/inductiva-sph-demos.git
```

Then, for analysis of the results, we suggest the installation of a few extra dependencies
specific to the demo:
```bash
$ cd inductiva-sph-demos
$ pip install -r requirements.txt
```
Some of the final visualizations require `ffmpeg`, an external dependency that needs
to be downloaded separately from [ffmpeg.org](https://ffmpeg.org/download.html) or
through Anaconda/Miniconda with `conda install -c conda-forge ffmpeg`.
````

sergiomsantos reviewed (commented):
some suggestions on the INSTALL and README files.
Others will follow

IvanPombo committed:
Add Install and ReadMe corrections by Sergio

IvanPombo committed:
Modify imports due to mixins

Yapf formatting

sergiomsantos commented on INSTALL.md:
````diff
@@ -0,0 +1,39 @@
+$ python -c 'import inductiva; print(inductiva.__version__)'
+<some version number printed here> 
+```
+#### Demo repository
````
missing line breaks:

````suggestion
The `inductiva` package is easily installed using `pip`:

```bash
$ pip install inductiva
```

A successful installation can be tested using:

```bash
$ python -c 'import inductiva; print(inductiva.__version__)'
<some version number printed here> 
```

#### Demo repository
````

sergiomsantos commented on INSTALL.md:
````diff
@@ -0,0 +1,39 @@
+```bash
+$ git clone https://github.com/inductiva/inductiva-sph-demos.git
+```
+Then, for analysis of the results, we suggest the installation of a few extra dependencies specific to the demos:
````
the reqs are not just for analysis. They are required.
Also, line to long

IvanPombo replied:
Forgot numpy was required and comes from pyvista.

In fact, I will add it, since we need it explicitly.

sergiomsantos commented on INSTALL.md:
````diff
@@ -0,0 +1,39 @@
+```bash
+$ cd inductiva-sph-demos
+$ pip install -r requirements.txt
+```
````
line break between blocks

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,19 @@
+</div>
 
-Here, we explore Smoothed Particle Hydrodynamics (SPH) with two simulators available via **Inductiva API** - [SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and [DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics). 
+With [**Inductiva API**](https://github.com/inductiva/inductiva/tree/main), exploring Smoothed Particle Hydrodynamics (SPH) simulations was never easier. 
```
line breaks; lets try to not go beyond the 100 char line lengths.

```suggestion
With [**Inductiva API**](https://github.com/inductiva/inductiva/tree/main), exploring
Smoothed Particle Hydrodynamics (SPH) simulations was never easier. 
```

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,19 @@
-In this repository, we make a tour of **Inductiva API**:
-- [Create a scenario](scenario/README.md);
-- [Generate a dataset via scenario](dataset/README.md).
+Users can launch the simulations they are familiar with through a simple Python interface, without having to worry about downloading/compiling and installing simulation packages and managing computational resources.
```
```suggestion
Users can launch the simulations they are familiar with through a simple Python interface,
without having to worry about downloading/compiling and installing simulation packages
and managing computational resources.
```

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,19 @@
+Users can launch the simulations they are familiar with through a simple Python interface, without having to worry about downloading/compiling and installing simulation packages and managing computational resources.
 
-### Install
+Two open-source SPH simulators are available via **Inductiva API** - [SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and [DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics). Users can use them directly or leverage the API to create custom scenarios and explore design spaces with simplicity.
```
```suggestion
Two open-source SPH simulators are available via **Inductiva API** -
[SplishSplash](https://github.com/inductiva/inductiva/wiki/SPlisHSPlasH) and
[DualSPHysics](https://github.com/inductiva/inductiva/wiki/DualSPHysics).
Users can use them directly or leverage the API to create custom scenarios and
explore design spaces with simplicity.
```

sergiomsantos commented on README.md:
````diff
@@ -1,31 +1,19 @@
-```
-
-To further learn about the `inductiva` package, check the [Inductiva API documentation](https://github.com/inductiva/inductiva/wiki).
+To learn more about the `inductiva` package, check the [Inductiva API documentation](https://github.com/inductiva/inductiva/wiki) and here to learn more about [SPH](https://en.wikipedia.org/wiki/Smoothed-particle_hydrodynamics).
````
line breaks

```suggestion
To learn more about the `inductiva` package, check the
[Inductiva API documentation](https://github.com/inductiva/inductiva/wiki) and the
[SPH Wikipedia page](https://en.wikipedia.org/wiki/Smoothed-particle_hydrodynamics)
to learn more about Smoothed Particle Hydrodynamics.
```

sergiomsantos commented on SPH_SIMULATIONS.md:
```diff
@@ -0,0 +1,65 @@
+## Running an SPH Simulation
+
+For **power users** who know how to configure a specific simulator and prepare the respective simulation files, **Inductiva API** provides a simple interface to simulate on a large scale. Here, we demonstrate how to use two SPH simulators in **Inductiva API**.
```
```suggestion
This section assumes users know how to configure a SplishSplash and/or DualPhysics simulation,
in which case our **Inductiva API client** provides a simple interface to run those simulations at
scale. Users are, otherwise, invited to consult the relevant documentation for each of those
simulation engines (LINKS HERE and HERE, respectively).
```

IvanPombo replied:
While I understand your point of setting  **Inductiva API client**, I think keeping it simple as it was is more clean and still includes everything.

Further, in the new line 5, `otherwise` seems contradictory. I think "Users are, in any case, invited..."

sergiomsantos reviewed (commented):
additional comments on README and INSTALL

IvanPombo committed:
Address Line breaks and Sergio comments

IvanPombo committed:
Changing the flow of FluidBlock

Clean the text formatting

sergiomsantos commented on README.md:
```diff
@@ -1,31 +1,28 @@
+
+Users can launch the simulations they are familiar with through a simple Python interface,
+without having to worry about downloading/compiling and installing simulation packages and
+managing computational resources.
```
```suggestion
managing/maintaining computational resources.
```

sergiomsantos commented on SPH_SIMULATIONS.md:
```diff
@@ -0,0 +1,82 @@
+The required one is a `.json` file that contains the parameters of the simulation.
+The others are extra input files that determine the geometry and properties of the
+objects and fluids in the simulation.
+
```
This section requires an introduction to the problem, ie, what are we simulation.
I suggest somethinf like:


```suggestion
### Falling fluid box with [SPlisHSPlasH](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH)

This example uses a straight forward call to the low-level api methods in `Inductiva API`
to simulate a cube of <fluid> falling inside an empty box. The fluid cube has size xxx and
is being dropped into a an empty cubic box with side YYY.  
Input to the simulator require one  `config.json` containing the configuration parameters
of the simulation and one `unit_box.obj` file defining the geometry on a unit cube (used
to describe the shape of both the falling fluid cube and container box).

 This example assumes all input files are inside a single folder, which gets passed to the
 `run` method of a `SPlisHSPlasH` instance as an argument, along with the name of the
 file configuring the simulation.
 Simulations are submitted as tasks to a remote execution queue. A free executor will then
 pick the task from the queue and run it with SPlisHSPlasH. Hence, the user needs to wait for
 the task to finish before retrieving the output files generated by the simulator.
 The following snippets exemplifies how this flow can be achieved.
 

```

sergiomsantos commented on SPH_SIMULATIONS.md:
````diff
@@ -0,0 +1,82 @@
+The others are extra input files that determine the geometry and properties of the
+objects and fluids in the simulation.
+
+```python
+import inductiva
+
+input_dir = inductiva.utils.files.download_from_url(
+    "https://storage.googleapis.com/inductiva-api-demo-files/"
+    "splishsplash-input-example.zip", unzip=True)
+
+# Set simulation input directory
+splishsplash = inductiva.simulators.SplishSplash()
+
+task = splishsplash.run(input_dir=input_dir,
+                        sim_config_filename="config.json")
+
+# Wait for the simulation to finish
+task.wait()
+task.download_outputs()
+```
````
````suggestion

1. Inspect the content of the input folder for this simulation:

```bash
$ ls -l1 sph_demos/simple_splishsplash
config.json
unit_box.obj
```

2. Inspect the content of the python script:

```python
$ cat simple_splishsplash.py
import inductiva

# Instantiate a SPlisHSPlasH proxy
splishsplash = inductiva.simulators.SplishSplash()

# submit the task to the remote execution pipeline, indicating 
# the path to the local directory containing the required input files
# and the name of the configuration file
task = splishsplash.run(input_dir='sph_demos/splishsplash',
                        sim_config_filename="config.json")

# wait for the simulation to finish
task.wait()

# and download all generated output files
task.download_outputs()
```

3. Running the python script:

```bash
$ python simple_splishsplash.py
2023-12-20 14:51:06 INFO Task ID: 1703083865802376474
2023-12-20 14:51:06 INFO Packing input archive to upload to inductiva.ai.
2023-12-20 14:51:06 INFO Uploading packed input archive with size 1.07 KB.
2023-12-20 14:51:06 INFO Input archive uploaded.
2023-12-20 14:51:06 INFO Task submitted to the default resource pool.
2023-12-20 14:51:21 INFO Task completed successfully.
2023-12-20 14:51:30 INFO Downloading simulation outputs to inductiva_output/1703083865802376474/output.zip.
100%|██████████████████████| 261k/261k [00:00<00:00, 5.15MB/s]
2023-12-20 14:51:31 INFO Uncompressing the outputs to inductiva_output/1703083865802376474.
```

4. Inspecting the output of the simulation in `inductiva_output/1703083865802376474`:

```bash
$ ls -l1 inductiva_output/1703083865802376474
log
stderr.txt
stdout.txt
vtk
```

````

sergiomsantos commented on SPH_SIMULATIONS.md:
```diff
@@ -0,0 +1,82 @@
+<img src="assets/simulation_logs.png" alt="Simulation Logs">
+
+
+### DualSPHysics
```
let's do something similar for this case.
Get all input files into a local folder `sph_demos/simple_dualsphysics` and generate a descritption of what the sim is doing.

sergiomsantos commented on SPH_SIMULATIONS.md:
````diff
@@ -0,0 +1,82 @@
+task.download_outputs()
+```
+
+This simulation is what is providing the amazing video of the duck floating in the water!
````
this sentence is misleading because we don't say how it can be done. Either remove it, or say how it can be done.

sergiomsantos commented on SPH_SIMULATIONS.md:
```diff
@@ -0,0 +1,82 @@
+    {"cmd": "partvtk -dirin floating_duck/data -savevtk floating_duck/particles/PartFluidOut", "prompts": []},
+    {"cmd": "isosurface -dirin floating_duck/data -saveiso floating_duck/Surface/surf", "prompts": []}]
+    
+task = simulator.run(input_dir="FloatingDuck",commands=commands)
+
+# Initialize the Simulator
+dualsphysics = inductiva.simulators.DualSPHysics()
+
+# Run simulation with config files in the input directory
+task = dualsphysics.run(input_dir=input_dir,
+                        commands=commands)
```
2 tasks?

sergiomsantos reviewed (commented):
some comments on SPH_SIMULATIONS

sergiomsantos commented on FLUID_BLOCK.md:
```diff
@@ -0,0 +1,312 @@
+
+<div align="center">
+    <img src="/assets/fluid_block.gif" width=500 alt="Fluid Block Movie">
+</div>
```
```suggestion
<div align="center">
    <img src="assets/fluid_block.gif" width=500 alt="Fluid Block Movie">
</div>
```

sergiomsantos commented on FLUID_BLOCK.md:
```diff
@@ -0,0 +1,312 @@
+# Fluid Block simulation
+
+Consider the following SPH simulation: a **fluid block** with certain physical properties
+is set free under the effect of gravity.
```
```
The examples in this section extend upon the straight forward API calls from the
[previous section](SPH_SIMULATIONS.md) and show how the user can abstract the
scientific problem into scenarios that use the API's robust templating mechanism to
dynamically generate input files for their simulations.

The examples herein aim at simulating the fluid dynamics of a falling fluid box into an empty
cubic container, with variable dimensions, initial position and velocity. The following clip
depicts the problem:
```

IvanPombo committed:
Add config files, low-level scripts, and more descriptive readme

Formatting correction

sergiomsantos commented on FLUID_BLOCK.md:
```diff
@@ -0,0 +1,312 @@
+# Fluid Block simulation
```
the order should be:
1. problem statement
2. the model (classes for block and fluids)
3. How models translate to input files - a bit about the FileManager and the rendering mechanism, without going into many details. Just some snippets to show how object attributes find their way into the files. The user can then go on and inspect deeper
4.  The simulators - talk a bit about the specific implementation details of the simulate method between both cases and how the same models can be used across both scenarios. Here we also talk a bit about the Simulation parameter class to configure the runs. Nothing to extended. The user will get the gist.
5. how to use the simulators and create a few dynamic runs with varying modesl/configs.
6.

IvanPombo replied:
I think the 2. and 3. points make sense, since looking at it again the templating intro seems cumbersome.

For the 4., I am not sure If we should highlight the construction for both simulators. It seems that we should just point to that.

The 5th wraps up the introduction, so I didn't want to extend to much.

sergiomsantos reviewed (commented):
check my comments on FLUID_BLOCK.md. We'll meet to discuss it further.

IvanPombo committed:
Clean the repository and improve the text of fluid block readme

sergiomsantos commented on 1_simple_dualsphysics.py:
```diff
@@ -0,0 +1,19 @@
+            "-onlytype:+fluid",
+     "prompts": []}]
+
+# Initialize the Simulator
+dualsphysics = inductiva.simulators.DualSPHysics()
+# Run simulation with config files in the input directory
+task = dualsphysics.run(input_dir="sph-example-dirs/simple-dualsphysics",
+                        commands=commands)
```
```suggestion
# Initialize the Simulator
dualsphysics = inductiva.simulators.DualSPHysics()

# Run simulation with config files in the input directory
task = dualsphysics.run(input_dir="sph-example-dirs/simple-dualsphysics",
                        commands=commands)
```

sergiomsantos commented on 3_generate_dataset.py:
```diff
@@ -0,0 +1,70 @@
+from lib import models
+from lib import scenarios
+
+# Constant parameters for all simulations
```
```suggestion
# Constant parameters for all simulations in this dataset
```

sergiomsantos commented on 3_generate_dataset.py:
```diff
@@ -0,0 +1,70 @@
+from lib import scenarios
+
+# Constant parameters for all simulations
+DENSITY = 1e3  # in kg/m^3
+TANK_DIMENSIONS = [1., 1., 1.]  # in meters
+SIMULATION_TIME = 3.  # in seconds
+PARTICLE_RADIUS = 0.008  # in meters
+OUTPUT_DIR = "dataset"
+N_SIMULATIONS = 10
```
align comments 
```suggestion
DENSITY = 1e3                              # in kg/m^3
TANK_DIMENSIONS = [1., 1., 1.]  # in meters
SIMULATION_TIME = 3.               # in seconds
PARTICLE_RADIUS = 0.008        # in meters
OUTPUT_DIR = "dataset"
N_SIMULATIONS = 10
```

sergiomsantos commented on docs/1_SPH_SIMULATIONS.md:
```diff
@@ -0,0 +1,148 @@
+## Running an SPH Simulation
```
80 char line width across all file

sergiomsantos reviewed (commented):
LGTM. Just minor formatting corrections

IvanPombo committed:
Last comments
